### PR TITLE
Feature #2460 allow missing input

### LINKED ISSUE #2460: Enhancement: Prevent error if some input files are missing
In the NOAA/METplus Telecon on 1/8/2024, we discussed adding a config option to allow runs to continue without error if some of the required input files are not found. This would prevent an error being reported when one of the runs did not succeed due to a data outage.

## Describe the Enhancement ##
Add a configuration option to report a warning instead of an error when input files are not found.
Shannon noted that she has written logic in her scripts that call METplus to check input files and only run if 80% of the input is available. The new config option could be a number between 0-1 that can be used to report warnings until a threshold is met, in which case errors will be throw. Alternatively, the error could be reported at the end.

### Time Estimate ###
*Estimate the amount of work required here.*
*Issues should represent approximately 1 to 3 days of work.*

### Sub-Issues ###
Consider breaking the enhancement down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
METplus 6.0.0

### Funding Source ###
2782544

### Assignee ###
- [ ] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [ ] Select **component(s)**
- [ ] Select **priority**
- [ ] Select **requestor(s)**

### Projects and Milestone ###
- [ ] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED CYCLE ASSIGNMENT** label
- [ ] Select **Milestone** as the next official version or **Future Versions**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [ ] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdataio](https://github.com/dtcenter/METdataio/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Enhancement Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issues
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by georgemccabe**
## Pull Request Testing ##

- [X] Describe testing already performed for these changes:</br>

Added unit tests for new capability to allow missing input files and still run without error
Ensured all existing tests and use cases pass

- [X] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

@JohnHalleyGotway and I met to discuss the details of this PR because there are a lot of changes. All tests should pass with no differences.

Review beta3 release notes: https://metplus--2493.org.readthedocs.build/en/2493/Users_Guide/release-notes.html#metplus-version-6-0-0-beta-3-release-notes-2024-02-07

- [X] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[No]**

Additional changes should be made after the beta3 release, including adding documentation for the new config variables. I created issue #2492 to complete remaining tasks. 

- [X] Do these changes include sufficient testing updates? **[Yes]**

Added and enhanced unit tests

- [X] Will this PR result in changes to the test suite? **[No]**</br>

- [X] Please complete this pull request review by **2/7/2024**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [X] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [X] Review the source issue metadata (required labels, projects, and milestone).
- [X] Complete the PR definition above.
- [X] Ensure the PR title matches the feature or bugfix branch name.
- [X] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [x] After submitting the PR, select the :gear: icon in the **Development** section of the right hand sidebar. Search for the issue that this PR will close and select it, if it is not already selected.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: docs/Users_Guide/release-notes.rst [hunk 1]
```diff
@@ -30,6 +30,49 @@ When applicable, release notes are followed by the
 `GitHub issue <https://github.com/dtcenter/METplus/issues>`__ number which
 describes the bugfix, enhancement, or new feature.
 
+METplus Version 6.0.0 Beta 3 Release Notes (2024-02-07)
```
Comment (JohnHalleyGotway) [inline docs/Users_Guide/release-notes.rst:33]:
```suggestion
METplus Version 6.0.0 Beta 3 Release Notes (2024-02-08)
```

File: docs/Users_Guide/release-notes.rst [hunk 2]
```diff
@@ -30,6 +30,49 @@ When applicable, release notes are followed by the
...
+-------------------------------------------------------
+
+  .. dropdown:: Enhancements
+
+     * Add suport for MET land-mask settings in Point-Stat
```
Comment (JohnHalleyGotway) [inline docs/Users_Guide/release-notes.rst:38]:
```suggestion
     * Add support for MET land-mask settings in Point-Stat
```

**Review by JohnHalleyGotway — APPROVED** [review]
I approve of this PR, as soon as we're able to get a clean GHA run of the testing workflow that does not flag any diffs.

Note that I proposed 2 minor edits in the release notes, updating the release date and fixing a typo. I did check all the issue links and confirmed that they do work.

As for the actual code changes, @georgemccabe walked me through them, and they make sense. But with the possibility of unintended consequences, we should get a clean run before merging.

**Comment by georgemccabe** [comment]
@JohnHalleyGotway, I will fix the typos in the release notes in the PR for creating the release.

**Pull request merged**